src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/UpdateVCFSequenceDictionary.java

```diff
@@ -167,27 +167,43 @@ public void closeTool() {
         }
     }
 
-    // either throws or returns a valid seq dict
-    private SAMSequenceDictionary getSequenceDictionaryFromInput(final String source) {
-        SAMSequenceDictionary dictionary;
-        if (source == null) {
-            if (hasReference()) {
-                dictionary = getReferenceDictionary();
+    // Override getBestAvailableSequenceDictionary() to ensure that the new sequence dictionary specified by
+    // the user (and not the sequence dictionary from the VCF we're trying to update) is used uniformly by all
+    // callers. Otherwise, the wrong dictionary would be used when writing the index for the output vcf.
+    //
+    @Override
+    public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+
+        SAMSequenceDictionary resultDictionary;
+
+        final SAMSequenceDictionary masterDictionary = getMasterSequenceDictionary();
+        if (dictionarySource == null) {
+            if (masterDictionary != null) {
+                // We'll accept the master dictionary if one was specified. Using the master dictionary
+                // arg will result in sequence dictionary validation.
```

Comment:
In the case where invalid dictionaries for the data are used (such as in #5087), can we detect it and throw a warning?  

If the user specifies a bad sequence dictionary we should let them know that something bad could happen.

(This is essentially the same request as the above comment.)

Reply:
The tool does have code to validate the new SD (in the first few lines of the apply method), and there are also tests for that. The problem in #5087 was that the indexer was getting the old (invalid) SD via the standard GATK machinery that uses the SD from the input. The change in this PR was to unify those so the new one is used everywhere.